include/sys/time_units.h

```diff
@@ -11,6 +11,19 @@
 extern "C" {
 #endif
 
+/** @brief System-wide macro to denote "forever" in milliseconds
+ *
+ *  Usage of this macro is limited to APIs that want to expose a timeout value
+ *  that can optionally be unlimited, or "forever".
+ *  This macro can not be fed into kernel functions or macros directly. Use
+ *  @ref SYS_MSEC instead.
+ */
+#define SYS_FOREVER_MS (-1)
```

Comment:
API meeting:
`SYS_FOREVER_MS` vs `SYS_FOREVER`

- Would we have additional `_S` and `_US` macros?
- Do we want the units specified in the macro?

Reply:
IMHO the `SYS_FOREVER_MS` is better option, it is more umambiguous than the one without MS
If really needed, _S and _US could be provided, but at least in networking side all the timeouts that need the value are in milliseconds.

Reply:
if such macro only exists for ms, then go for SYS_FOREVER_MS. But if other units would need this as well, I'd prefer SYS_FOREVER